tests/core/nodes/test_tools.py

```
@@ -0,0 +1,302 @@
+
+    def test_empty_list(self):
+        """Empty list produces no tools."""
+        tools, executor = tools_from_nodes([])
+
+        assert len(tools) == 0
+        assert callable(executor)
+
+    @pytest.mark.asyncio
+    async def test_executor_unknown_tool(self, bash_node):
+        """Executor returns error for unknown tool."""
+        tools, executor = tools_from_nodes([bash_node])
```

Ruff (0.14.10)

Unpacked variable `tools` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)